### ISSUE TITLE
Wasserstein discriminant analysis (WDA) with stochastic gradient descent

### ISSUE THREAD

**Issue by palVJ** [PythonOT/POT#665]
Thanks very much for this brilliant repo making optimal transport analysis easier for many!

I am wondering whether WDA is possible to use with stochastic gradient descent (SGD). I ask because I am working with a very large high-dimensional data set. Based on the information here: https://pythonot.github.io/gen_modules/ot.dr.html#id15 
I am not sure whether this is possible as of today, but I may be wrong.

The way I understand it, SGD should be applied by choosing the appropriate solver which should be of type pymanopt.optimizers (previously pymanopt.solvers). However based on comment here https://github.com/pymanopt/pymanopt/issues/60 it has been worked on by individuals, but not implemented in general in the pymanopt package, so one would need to manually do the necessary changes. 

Is my understanding correct so far, or am I missing something?

Cheers

**Comment by rflamary** [comment]
Yes it is no implemented in POT but you might be able to use SGD to solve WDA with gradient computation on batches using functions from POT. Still note that I have no idea how the gradient of the ratio of batches is biased wrt the true WDA ratio that is optimized.
